src/org/omegat/filters3/xml/XMLFilter.java

```diff
@@ -74,10 +78,26 @@ public abstract class XMLFilter extends AbstractFilter implements Translator {
     /** Creates a new instance of XMLFilter */
     public XMLFilter(XMLDialect dialect) {
         parserFactory = SAXParserFactory.newInstance();
-        // parserFactory.setValidating(false);
         try {
-            parserFactory.setFeature("http://xml.org/sax/features/validation", true);
-        } catch (Exception ignored) {
+            // We do not validate XML in default
+            // When driver writer want not to validate please override and set features false.
+            // ex. setSAXFeature("http://xml.org/sax/features/validation", false);
+
+            // protect from XXE attack
+            // "Feature for Secure Processing (FSP)" is the central mechanism to
+            // help safeguard XML processing. It instructs XML processors, such as parsers,
+            // validators, and transformers, to try and process XML securely.
+            parserFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            // Avoid internet connection to validate with external DTD.
+            parserFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+            // Disable external general entities
+            parserFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            // Disable external parameter entities
+            parserFactory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+            // as well, per Timothy Morgan's 2014 paper: "XML Schema, DTD, and Entity Attacks"
+            parserFactory.setXIncludeAware(false);
+        } catch (Exception ex) {
+            Log.log(ex.getMessage());
```

Comment:
It might be better to write a meaningful error message when an error occurs while trying to parse the XML file:

```suggestion
        } catch (ParserConfigurationException | SAXNotRecognizedException | SAXNotSupportedException ex) {
            Log.logErrorRB(ex, "XML_FILTER_ERROR", ex.getMessage());
        }
```

and in `Bundle.properties` (~line 1296):

```properties
XML_FILTER_ERROR=Unable to parse source XML file
```

Reply:
Thank you for feedback . adjusted.